diffrax/_solver/implicit_euler.py

```diff
@@ -81,14 +82,15 @@ def step(
         # write out a `ButcherTableau` and use `AbstractSDIRK`.
         k0 = terms.vf_prod(t0, y0, args, control)
         args = (terms.vf_prod, t1, y0, args, control)
-        nonlinear_sol = optx.root_find(
-            _implicit_relation,
-            self.root_finder,
-            k0,
-            args,
-            throw=False,
-            max_steps=self.root_find_max_steps,
-        )
+        with jax.numpy_dtype_promotion("standard"):
+            nonlinear_sol = optx.root_find(
```

Comment:
I think this one and similar won't be needed after https://github.com/patrick-kidger/optimistix/pull/53
I'll check it later

Reply:
Okay, lmk when you want a review of this!

Can you rebase on top of the latest `dev`? (We just merged a large change introducing stochastic Runge--Kutta methods. As this PR is small I'll try to get it in before the next large change... :D )